test/repository/hbase/unitlinks/UnitLinksNoPeriodRowMapperSpec.scala

```diff
@@ -13,83 +13,109 @@ class UnitLinksNoPeriodRowMapperSpec extends FreeSpec with Matchers {
 
-    val ParentsMapStr = Map(
+    val ParentsMap = Map(
       ParentPrefix + Enterprise -> SampleEnterpriseParentId
     )
-    val ChildrenMapStr = Map(
+    val ChildrenMap = Map(
       ChildPrefix + SampleCompaniesHouseChildId -> CompaniesHouse,
       ChildPrefix + SamplePayAsYouEarnChildId -> PayAsYouEarnTax
     )
-    val FamilyMapStr = ParentsMapStr ++ ChildrenMapStr
+    val FamilyMap = ParentsMap ++ ChildrenMap
+    val EditedMap = Map("edited" -> "Y")
   }
 
   "A UnitLinksNoPeriodRowMapper" - {
     "can create a UnitLinksNoPeriod from a row" - {
-      "when all fields are available and valid" in new Fixture {
-        val row = Row(RowKey, fields = FamilyMapStr)
+      "when all fields define either parents or children" in new Fixture {
+        val row = Row(RowKey, fields = FamilyMap)
+
+        UnitLinksNoPeriodRowMapper.fromRow(row) shouldBe Some(SampleUnitLinksNoPeriodWithAllFields)
+      }
+
+      "ignoring fields that do not define parents or children" in new Fixture {
+        val row = Row(RowKey, fields = FamilyMap ++ EditedMap)
 
         UnitLinksNoPeriodRowMapper.fromRow(row) shouldBe Some(SampleUnitLinksNoPeriodWithAllFields)
       }
 
       "when there are no children" in new Fixture {
-        val row = Row(RowKey, fields = ParentsMapStr)
+        val row = Row(RowKey, fields = ParentsMap)
 
         UnitLinksNoPeriodRowMapper.fromRow(row) shouldBe Some(SampleUnitLinksNoPeriodWithAllFields.copy(children = None))
       }
 
       "when there are no parents" in new Fixture {
-        val row = Row(RowKey, fields = ChildrenMapStr)
+        val row = Row(RowKey, fields = ChildrenMap)
 
         UnitLinksNoPeriodRowMapper.fromRow(row) shouldBe Some(SampleUnitLinksNoPeriodWithAllFields.copy(parents = None))
       }
+
+      "when there are neither parents or children" in new Fixture {
+        val row = Row(RowKey, fields = Map.empty)
+
+        UnitLinksNoPeriodRowMapper.fromRow(row) shouldBe Some(SampleUnitLinksNoPeriodWithAllFields.copy(parents = None, children = None))
+      }
     }
 
     "cannot create a UnitLinksNoPeriod from a row" - {
       "when the row key is invalid" - {
         "because it contains too few elements" in new Fixture {
-          val row = Row(rowKey = UnitIdStr, fields = FamilyMapStr)
+          val row = Row(rowKey = UnitIdStr, fields = FamilyMap)
 
           UnitLinksNoPeriodRowMapper.fromRow(row) shouldBe None
         }
 
         "because it contains too many elements" in new Fixture {
-          val row = Row(rowKey = s"$RowKey~201801", fields = FamilyMapStr)
+          val row = Row(rowKey = s"$RowKey~201801", fields = FamilyMap)
 
           UnitLinksNoPeriodRowMapper.fromRow(row) shouldBe None
         }
 
         "because it contains an unrecognised unit type" in new Fixture {
-          val row = Row(rowKey = s"UNKNOWN~$UnitIdStr", fields = FamilyMapStr)
+          val row = Row(rowKey = s"UNKNOWN~$UnitIdStr", fields = FamilyMap)
```

Comment:
That scenario is covered (albeit somewhat indirectly) by the test above:
* because it contains too few elements

i.e. if all the ~'s were replaced with _ - then it would be treated as a single component without the required number of elements.